src/utils/prompt.cc

```diff
@@ -101,26 +120,76 @@ void PromptOptions::setOptionHelp( unsigned opt, const std::string & help_str )
   _opt_help[opt] = help_str;
 }
 
-
-int PromptOptions::getReplyIndex( const std::string & reply ) const
+std::vector<int> PromptOptions::getReplyMatches( const std::string & reply_r ) const
 {
-  DBG << " reply: " << reply << " (" << str::toLower(reply) << " lowercase)" << endl;
+  std::vector<int> ret;
 
-  unsigned i = 0;
-  for( StrVector::const_iterator it = _options.begin(); it != _options.end(); ++i, ++it )
+  // #NUM ? (direct index into option vector)
+  if ( reply_r[0] == '#' && reply_r[1] != '\0' )
   {
-    DBG << "index: " << i << " option: " << *it << endl;
-    if ( *it == str::toLower(reply) )
+    unsigned num = 0;	// -1 : if no match
+    for ( const char * cp = reply_r.c_str()+1; *cp; ++cp )
```

Comment:
for loop could stop once num gets bigger than what was expected, in case someone types a really large number

Reply:
The intention is actually to catch and eat `/^#[0-9]+$/` and let everything else fall through to the string matcher. So I have to parse it to the end and anyway. Discarding `#NUM` does not depend on whether `NUM` matches the number of options (otherwise one could name an option literaly `#13` and this would work until >=13 options are used).